core/src/main/kotlin/org/evomaster/core/output/naming/NumberedTestCaseNamingStrategy.kt

```diff
@@ -21,18 +22,51 @@ open class NumberedTestCaseNamingStrategy(
     }
 
     // numbered strategy will not expand the name unless it is using the namingHelper
-    override fun expandName(individual: EvaluatedIndividual<*>, nameTokens: MutableList<String>): String {
+    override fun expandName(individual: EvaluatedIndividual<*>, nameTokens: MutableList<String>, ambiguitySolver: ((Action) -> List<String>)?): String {
         return ""
     }
 
+    override fun resolveAmbiguity(individualToName: MutableMap<EvaluatedIndividual<*>, String>, duplicatedIndividuals: MutableSet<EvaluatedIndividual<*>>) {
+        // do nothing, plain numbered strategy will never have duplicate names
+    }
+
     // kicking off with an empty mutableListOf for each test case to accumulate their own name tokens
     private fun getName(counter: Int, individual: EvaluatedIndividual<*>): String {
         return "test_${counter}${expandName(individual, mutableListOf())}"
     }
 
+    private fun concatName(counter: Int, expandedName: String): String {
+        return "test_${counter}${expandedName}"
+    }
+
     private fun generateNames(individuals: List<EvaluatedIndividual<*>>) : List<TestCase> {
+        val individualToName = mutableMapOf<EvaluatedIndividual<*>, String>()
```

Comment:
should not have mutable objects (eg `EvaluatedIndividual<*>`) as keys for a map. if for any reason a key is modified, then the map can become unusable

Reply:
maybe better something like:
```
val individualToName = individuals.map{Pair(it, expandName(it))}
```

Reply:
I understand, but the fact is that I need to have a way to bind the `EvaluatedIndividual` with the generated test case name for the latter to be replaced if it needs to be disambiguated. So, eventually I'll need something along the lines of a Map.
The latest commit, does not forward the `individualToName` to the following method, but just modifies it in the scope of the `generateNames` method. Maybe that's a better solution.

I see no reason for an `EvaluatedIndividual` to be modified while the test case names are being generated. That said, a `Lazy.assert { individuals.size == individualToName.size }` could be added just after the `getDuplicateNames(individualToName).forEach ...` call. So as to be sure nobody in the future changes an EvaluatedIndividual and therefore adds an extra element to the map. What do you think?

Reply:
@Pgarrett : i guess it can be ok here, as the map is not exposed (eg returned from a public method). 
as a rule of thumb, mutable keys should be avoided, and their map never exposed. 
debugging weird bugs related to a corrupted map (eg buckets based on hascode indices, but hashcode of stored elements has changed because their state has changed) is no fun... it is the kind of a bug that is really hard to track down :(